enterprise/server/scheduling/task_router/task_router.go

```diff
@@ -195,15 +260,66 @@ func (tr *taskRouter) routingKey(ctx context.Context, cmd *repb.Command, remoteI
 	return strings.Join(parts, "/"), nil
 }
 
-func copyNodes(nodes []interfaces.ExecutionNode) []interfaces.ExecutionNode {
-	out := make([]interfaces.ExecutionNode, len(nodes))
-	copy(out, nodes)
-	return out
+func (s runnerRecyclingRoutingStrategy) routingInfo(params routingParams) (int, string, error) {
+	nodeLimit := s.preferredNodeLimit(params)
+	key, err := s.routingKey(params)
+	return nodeLimit, key, err
 }
 
-func minInt(a, b int) int {
-	if a < b {
-		return a
+type outputAffinityRoutingStrategy struct {
+}
+
+func (outputAffinityRoutingStrategy) applies(params routingParams) bool {
+	if !platform.IsTrue(platform.FindValue(params.cmd.GetPlatform(), platform.OutputAffinityRoutingPropertyName)) {
+		return false
 	}
-	return b
+	return (len(params.cmd.OutputPaths) > 0 && params.cmd.OutputPaths[0] != "") ||
+		(len(params.cmd.OutputFiles) > 0 && params.cmd.OutputFiles[0] != "") ||
+		(len(params.cmd.OutputDirectories) > 0 && params.cmd.OutputDirectories[0] != "")
+}
+
+func (outputAffinityRoutingStrategy) preferredNodeLimit(params routingParams) int {
+	return defaultPreferredNodeLimit
+}
+
+func (outputAffinityRoutingStrategy) routingKey(params routingParams) (string, error) {
+	parts := []string{"task_route", params.groupID}
+
+	if params.remoteInstanceName != "" {
+		parts = append(parts, params.remoteInstanceName)
+	}
+
+	platform := params.cmd.GetPlatform()
+	if platform == nil {
+		platform = &repb.Platform{}
+	}
+	b, err := proto.Marshal(platform)
+	if err != nil {
+		return "", status.InternalErrorf("failed to marshal Command: %s", err)
+	}
+	parts = append(parts, fmt.Sprintf("%x", sha256.Sum256(b)))
```

Comment:
you can use hash.String() for this instead of sha256.Sum256

Reply:
Whoops, thanks